#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: <your description>

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- src/libstdc++-v3/config/locale/uclibc/c_locale.h.old	2010-07-10 14:38:23.082478073 -0400
+++ src/libstdc++-v3/config/locale/uclibc/c_locale.h	2010-07-10 16:00:50.463323922 -0400
@@ -108,7 +108,7 @@
 #endif
 
       va_start(__args, __fmt);
-      const int __ret = std::vsnprintf(__out, __size, __fmt, __args);
+      const int __ret = vsnprintf(__out, __size, __fmt, __args);
       va_end(__args);
 
 #ifdef __UCLIBC_HAS_XCLOCALE__
